### PR TITLE
Feature/smncopynumbercaller

### LINKED ISSUE #1290: Include SMNCopyNumberCaller
**Is your feature request related to a problem? Please describe.**
We have heard the rumours and been waiting eagerly, but now it's official: 
[SMNCopyNumberCaller](https://github.com/Illumina/SMNCopyNumberCaller). Good-enough:ish performance - really nice, given the difficulty level of doing this with short reads. See [preprint](https://www.medrxiv.org/content/10.1101/19006635v2).

**Describe the solution you'd like**
Include in MIP. Export annotation for Scout - need to figure out if we should load the json directly, or e.g. convert to vcf.

### PR DESCRIPTION
**Pull request by henrikstranneheim**
### This PR fixes:

- Adds SMNCopyNumberCaller to MIP Fix #1290

### How to test:

- Automatic and continuous test pass

### Expected outcome:
- Installation, unit and integration tests pass

### Review:
- [x] Code review
- [x] New code is executed and covered by tests
- [x] Tests pass

### CODE REVIEW

File: lib/MIP/Cli/Mip/Analyse/Rd_dna.pm [hunk 1]
```diff
@@ -665,6 +665,16 @@ q{Default: grch37_dbsnp_-138-.vcf, grch37_1000g_indels_-phase1-.vcf, grch37_mill
         )
     );
 
+    option(
```
Comment (jemten) [inline lib/MIP/Cli/Mip/Analyse/Rd_dna.pm]: Version bump this file

File: lib/MIP/Recipes/Analysis/Smncopynumbercaller.pm [hunk 2]
```diff
@@ -0,0 +1,288 @@
...
+                parameter_href         => $parameter_href,
+                recipe_name            => $recipe_name,
+            }
+        )
+    );
+
+    my $outdir_path_prefix  = $io{out}{dir_path_prefix};
+    my $outfile_name_prefix = $io{out}{file_name_prefix};
+    my $outfile_path        = $io{out}{file_path};
+    my $outfile_path_prefix = $io{out}{file_path_prefix};
```
Comment (jemten) [inline lib/MIP/Recipes/Analysis/Smncopynumbercaller.pm]: This variable isn't used anywhere

File: lib/MIP/Recipes/Analysis/Smncopynumbercaller.pm [hunk 3]
```diff
@@ -0,0 +1,288 @@
...
+        set_recipe_outfile_in_sample_info(
+            {
+                infile           => $outfile_name_prefix,
+                path             => $outfile_path,
+                recipe_name      => $recipe_name,
+                sample_id        => $sample_id,
+                sample_info_href => $sample_info_href,
+            }
+        );
+
```
Comment (jemten) [inline lib/MIP/Recipes/Analysis/Smncopynumbercaller.pm]: Should we store this file?

File: lib/MIP/Recipes/Analysis/Smncopynumbercaller.pm (hunk 2 above)
Comment (henrikstranneheim) [reply lib/MIP/Recipes/Analysis/Smncopynumbercaller.pm]: removed

File: lib/MIP/Cli/Mip/Analyse/Rd_dna.pm (hunk 1 above)
Comment (henrikstranneheim) [reply lib/MIP/Cli/Mip/Analyse/Rd_dna.pm]: Done

**Pull request merged**